test/e2e/marketplace_test.go

```diff
@@ -132,27 +134,31 @@ func defaultCreateTest(t *testing.T, f *test.Framework, ctx *test.TestCtx) error
 		return err
 	}
 
-	// Check for the config map created from the catalogsourceconfig.
-	resultConfigMap := &corev1.ConfigMap{}
-	err = WaitForResult(t, f, resultConfigMap, namespace, configMapName)
+	// Then check for the catalog source.
+	resultCatalogSource := &olm.CatalogSource{}
+	err = WaitForResult(t, f, resultCatalogSource, namespace, catalogSourceName)
 	if err != nil {
 		return err
 	}
 
-	// Then check for the catalog source.
-	resultCatalogSource := &olm.CatalogSource{}
-	err = WaitForResult(t, f, resultCatalogSource, namespace, catalogSourceName)
+	// Then check that the service was created.
+	resultService := &corev1.Service{}
+	err = WaitForResult(t, f, resultService, namespace, serviceName)
 	if err != nil {
 		return err
 	}
 
-	// Assert that the catalogsource spec properly references the configmap.
-	if resultCatalogSource.Spec.ConfigMap != resultConfigMap.Name {
-		t.Errorf(
-			"The created catalogsource %s was not properly associated with the created configmap %s",
-			resultCatalogSource.Name,
-			resultConfigMap.Name,
-		)
+	// Then check that the deployment was created.
+	resultDeployment := &apps.Deployment{}
+	err = WaitForResult(t, f, resultDeployment, namespace, deploymentName)
+	if err != nil {
+		return err
+	}
+
+	// Now check that the deployment is ready.
+	err = WaitForSuccessfulDeployment(t, f, *resultDeployment)
```

Comment:
In an asynchronous world, we would return as soon as we successfully create the Deployment object. We can watch the `pods` and other relevant resources with appropriate labels to get to the desired state. We can address this in the next iteration.